src/python/pants/backend/python/lint/black/rules.py

```diff
@@ -87,69 +48,77 @@ def create_execute_request(
       entry_point=black.get_entry_point(),
     )
   )
+  return BlackSetup(requirements_pex=requirements_pex, config_snapshot=config_snapshot)
 
-  sources_digest = wrapped_target.target.sources.snapshot.directory_digest
 
-  merged_input_files = await Get(
-    Digest,
-    DirectoriesToMerge(
-      directories=(
-        sources_digest,
-        resolved_requirements_pex.directory_digest,
-        config_snapshot.directory_digest,
-      )
-    ),
-  )
-  return BlackSetup(config_path, resolved_requirements_pex, merged_input_files)
+@dataclass(frozen=True)
+class BlackArgs:
+  args: Tuple[str, ...]
```

Comment:
Introducing this `dataclass`, rather than just having a `black_args: Tuple[str, ...]` on the `BlackSetup` seems like overkill, but I don't mind much :)

Reply:
The reason for this is that this changes `BlackSetup` to solely have the setup for Black that can be used across any target. If we had the args attached to it there, we would need a new `BlackSetup` for every target, because the args include the names of the files to run against. Instead, this results in one `BlackSetup` for everything and 1-1 `BlackArgs->target`.

Reply:
Aha, I thought this was only used in `BlackSetup` but it's used somewhere else :) Never mind!